### PR TITLE
Return key from older Fixnum class earlier

### PR DESCRIPTION
**Pull request by ashmaroli**
return both key of Ruby24 `Integer` class and Ruby 23- `Fixnum` class earlier

### CODE REVIEW

File: lib/wab/impl/data.rb [hunk 1]
```diff
@@ -375,14 +375,13 @@ def values_eql?(v0, v1)
...
+        return key if (key.is_a?(Integer) || WAB::Utils.pre_24_fixnum?(key))
 
         key = key.to_s if key.is_a?(Symbol)
         if key.is_a?(String)
           i = key.to_i
           return i if i.to_s == key
         end
-        return key if WAB::Utils.pre_24_fixnum?(key)
 
         raise StandardError.new("path key must be an integer for an Array.") unless i.to_s == key
```
Comment (ashmaroli) [inline lib/wab/impl/data.rb]: since `i` is defined in the `if` block at ln#381, would it be still available for the `raise` condition?

Comment (ohler55) [inline lib/wab/impl/data.rb]: It would not. Actually, the ```unless i.to_s == key``` should be removed. That is a bug. A remnant of changing the code.

Comment (ashmaroli) [inline lib/wab/impl/data.rb]: Ah! You can fix it directly on the `develop` branch then :+1:

**Comment by ohler55** [comment]
Fixed it.

**Comment by ashmaroli** [comment]
But then, this PR is not related to that fix.
This PR moves **another `raise` call** higher up in the track

File: lib/wab/impl/data.rb (hunk 1 above)
Comment (ohler55) [inline lib/wab/impl/data.rb]: Oh, I see. I think I would rather keep the pre 2.4 lower in the checks as I would hope people elect to use the latest 2.4 or above. It matter less if older versions are slower. Better to keep the target faster and let the older ones have a reason to move up.

Comment (ashmaroli) [inline lib/wab/impl/data.rb]: Sounds reasonable.. :+1:

**Pull request closed without merging**